.github/workflows/sync_typeshed.yaml

```diff
@@ -46,10 +46,19 @@ jobs:
           cp -r typeshed/stdlib ruff/crates/red_knot_vendored/vendor/typeshed/stdlib
           rm -rf ruff/crates/red_knot_vendored/vendor/typeshed/stdlib/@tests
           git -C typeshed rev-parse HEAD > ruff/crates/red_knot_vendored/vendor/typeshed/source_commit.txt
+
           # Patch the typeshed stubs to include `knot_extensions`
-          ln -s ../../../knot_extensions/knot_extensions.pyi ruff/crates/red_knot_vendored/vendor/typeshed/stdlib/
-          echo "# Patch applied for red_knot:" >> ruff/crates/red_knot_vendored/vendor/typeshed/stdlib/VERSIONS
-          echo "knot_extensions: 3.0-" >> ruff/crates/red_knot_vendored/vendor/typeshed/stdlib/VERSIONS
+          source="crates/red_knot_vendored/knot_extensions/knot_extensions.pyi"
+          target="crates/red_knot_vendored/vendor/typeshed/stdlib/knot_extensions.pyi"
+          (
+            echo "# WARNING: This file is generated by the typeshed sync workflow. Do not edit it manually."
+            echo "# Edit the source file at '$source' instead and then copy it over to this file."
+          ) > "ruff/$target"
+          cat "ruff/$source" >> "ruff/$target"
+          (
+            echo "# Patch applied for red_knot:"
+            echo "knot_extensions: 3.0-"
+          ) >> ruff/crates/red_knot_vendored/vendor/typeshed/stdlib/VERSIONS
```

Comment:
I don't mind the approach taken here. An alternative could be to copy the file in the `red_knot_vendored`'s `build.rs`.

Reply:
Ohhh, I like this much more. Thanks!